src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = Object.assign({}, state);
+  const arrState = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(copyState, extraData);
+        break;
+      case 'removeProperties':
```

Comment:
to make everything look prettier, I'd add some whitespaces over my code where needed